Example: how to manually serialize and deserialize Fn in struct

Arrow Pull Request: https://github.com/DSLAM-UMD/arrow/pull/1

```rust
/// Physical expression of a scalar function
#[derive(Serialize, Deserialize)]
pub struct ScalarFunctionExpr {
    fun: ScalarFunctionImplementation,
    name: String,
    args: Vec<Arc<dyn PhysicalExpr>>,
    return_type: DataType,
}

/// Scalar function
pub type ScalarFunctionImplementation =
    Arc<dyn Fn(&[ArrayRef]) -> Result<ArrayRef> + Send + Sync>;
```

```rust
error[E0277]: the trait bound `dyn for<'r> Fn(&'r [Arc<(dyn arrow::array::Array + 'static)>]) -> std::result::Result<Arc<(dyn arrow::array::Array + 'static)>, DataFusionError> + Sync + std::marker::Send: physical_plan::_::_serde::Serialize` is not satisfied
   --> datafusion/src/physical_plan/functions.rs:391:5
    |
391 |     fun: ScalarFunctionImplementation,
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `physical_plan::_::_serde::Serialize` is not implemented for `dyn for<'r> Fn(&'r [Arc<(dyn arrow::array::Array + 'static)>]) -> std::result::Result<Arc<(dyn arrow::array::Array + 'static)>, DataFusionError> + Sync + std::marker::Send`
    |
    = note: required because of the requirements on the impl of `physical_plan::_::_serde::Serialize` for `Arc<dyn for<'r> Fn(&'r [Arc<(dyn arrow::array::Array + 'static)>]) -> std::result::Result<Arc<(dyn arrow::array::Array + 'static)>, DataFusionError> + Sync + std::marker::Send>`
```

--- a/examples/Cargo.toml
+++ b/examples/Cargo.toml
@@ -16,8 +16,8 @@ serde_with = { version = "1.6.0", features = ["json"]}
 aws_lambda_events = "0.3"
 base64 = "0.13"
 more-asserts = "0.2.1"
-arrow = { git = "https://github.com/apache/arrow", branch = "master" }
-datafusion = { git = "https://github.com/apache/arrow", branch = "master" }
+arrow = { git = "https://github.com/DSLAM-UMD/arrow", branch = "scq" }
+datafusion = { git = "https://github.com/DSLAM-UMD/arrow", branch = "scq" }
 typetag = "0.1"
 serde_traitobject = "0.2.7"
 serde_closure = "0.3.2"

--- a/src/common/Cargo.toml
+++ b/src/common/Cargo.toml
@@ -10,4 +10,4 @@ autobins = false
 
 [dependencies]
 sqlparser = "0.6.1"
-arrow = { git = "https://github.com/apache/arrow", branch = "master" }
+arrow = { git = "https://github.com/DSLAM-UMD/arrow", branch = "scq" }

--- a/src/legacy/datasource/Cargo.toml
+++ b/src/legacy/datasource/Cargo.toml
@@ -9,7 +9,7 @@ autobins = false
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
 [dependencies]
-arrow = { git = "https://github.com/apache/arrow", branch = "master" }
+arrow = { git = "https://github.com/DSLAM-UMD/arrow", branch = "scq" }
 common = {path = "../common"}
 physical_plan = {path = "../physical_plan"}
 tokio = { version = "0.2", features = ["macros"] }

--- a/src/legacy/logical_plan/Cargo.toml
+++ b/src/legacy/logical_plan/Cargo.toml
@@ -10,5 +10,5 @@ autobins = false
 
 [dependencies]
 sqlparser = "0.6.1"
-arrow = { git = "https://github.com/apache/arrow", branch = "master" }
+arrow = { git = "https://github.com/DSLAM-UMD/arrow", branch = "scq" }
 common = {path = "../common"}

--- a/src/legacy/physical_plan/Cargo.toml
+++ b/src/legacy/physical_plan/Cargo.toml
@@ -17,4 +17,4 @@ rusoto_core = "0.45.0"
 rusoto_kinesis = "0.45.0"
 serde_json = "1.0"
 common = {path = "../common"}
-arrow = { git = "https://github.com/apache/arrow", branch = "master" }
+arrow = { git = "https://github.com/DSLAM-UMD/arrow", branch = "scq" }

